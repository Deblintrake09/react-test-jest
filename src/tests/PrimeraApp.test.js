import React from "react";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe('Pruebas de PrimeraApp',()=>{

   /* test('debe mostar el mensaje "Hola, Soy Batman!!"',()=>{
        const saludo = "Hola, Soy Batman!!";
        const wrapper = render(<PrimeraApp saludo={saludo}/>)

        expect(wrapper.getByText(saludo)).toBeInTheDocument();
    })*/

    test('debe mostar <PrimeraApp/> correctamente', ()=>{
        const saludo = "Hola, Soy Batman!!";
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostarr el subtitulo pasado por props',()=>{ 
        const saludo = "Hola, Soy Batman!!";
        const subtitulo = "Oscuridad! Sin Padres!"
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo}/>)

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subtitulo);
    });
})