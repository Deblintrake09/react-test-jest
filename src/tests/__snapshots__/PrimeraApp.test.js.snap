// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de PrimeraApp debe mostar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Batman!!
  </h1>
  <p>
     Soy un Subtitulo
  </p>
</Fragment>
`;
